pkg/funky/server.go

```diff
@@ -0,0 +1,174 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+///////////////////////////////////////////////////////////////////////
+package funky
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+// Server an interface for managing function servers
+type Server interface {
+	IsIdle() bool
+	SetIdle(bool)
+	GetPort() uint16
+	Invoke(input Message) (io.ReadCloser, error)
+	Stdout() *bytes.Buffer
+	Stderr() *bytes.Buffer
```

Comment:
Stdout() and Stderr() should probably have a different return type: either `[]byte` or `io.Reader`.

The implementation can still use `*bytes.Buffer`. 

Also, these two methods should only return the content of stdout/stderr related to a single invocation. So, maybe it makes sense to return an `Invocation` object as a result of `Invoke()` call, which would have the responsibilities of actually running the invocation and returning the related stderr/stdout.